### PR TITLE
Add Partial Tag Helper 2.1 doc

### PR DESCRIPTION
**Pull request by scottaddie**
Fixes https://github.com/aspnet/Docs/issues/5420

This PR also upgrades the shared Tag Helpers sample app to 2.1.

[Internal Review Page](https://review.docs.microsoft.com/en-us/aspnet/core/mvc/views/tag-helpers/built-in/partial-tag-helper?branch=pr-en-us-5533)

### CODE REVIEW

File: aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ms.topic: article
+uid: mvc/views/tag-helpers/builtin-th/partial-tag-helper
+---
+# Partial Tag Helper
+
+By [Scott Addie](https://github.com/scottaddie)
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/mvc/views/tag-helpers/built-in/samples) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+The Partial Tag Helper is supported as of ASP.NET Core 2.1. It offers an alternative to [HTML Helper syntax](xref:mvc/views/partial#referencing-a-partial-view) for rendering a [partial view](xref:mvc/views/partial) in Razor Pages or MVC.
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md]: The first paragraph needs to say what it does.

Comment (Rick-Anderson) [reply aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md]: > It offers an alternative to [HTML Helper syntax](xref:mvc/views/partial#referencing-a-partial-view) for rendering a [partial view](xref:mvc/views/partial) in Razor Pages or MVC.

Use wording from https://docs.microsoft.com/en-us/aspnet/core/mvc/views/working-with-forms

Something like the HTML Helpers alternative is

File: aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+An inventory of this Tag Helper's attributes follows.
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md]: I'd make this an H2

## Attributes 
then the others are H3's

File: aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md (hunk 1 above)
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md]: I think this works better as a list:

* require 2.1 or later.
* has an HTML alternative ...

File: aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+[!code-cshtml[](samples/TagHelpersBuiltIn/Pages/Product.cshtml?name=snippet_AspFor)]
+
+## view-data
+
+The `view-data` attribute assigns a [ViewDataDictionary](/dotnet/api/microsoft.aspnetcore.mvc.viewfeatures.viewdatadictionary) to pass to the partial view. The following markup makes the entire ViewData collection accessible to the partial view:
+
+[!code-cshtml[](samples/TagHelpersBuiltIn/Pages/Product.cshtml?name=snippet_ViewData&highlight=5-)]
+
+Notice that the `IsNumberReadOnly` key value is set to `true` and added to the ViewData collection. Consequently, `ViewData["IsNumberReadOnly"]` is made accessible within the partial view:
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md]: Move this below the next snippet. I was looking for `IsNumberReadOnly` in the preceding code.

I'd use wording like:

In the preceding markup,  the `IsNumberReadOnly` key value is set to `true` and added to the ViewData collection. Consequently, 

Not sure you need Consequently but OK to leave it.

**Comment by NTaylorMullen** [comment]
We have a few key changes coming to PartialTagHelper in the preview2 release. Should we hold off on this document until that's completed?

- https://github.com/aspnet/Mvc/issues/7379
- https://github.com/aspnet/Mvc/issues/7374

**Comment by scottaddie** [comment]
@NTaylorMullen I'd prefer to publish a doc for folks using Preview1. We took this approach with the new SPA templates and got some really great early feedback as a result. With that in mind, is this doc accurate for Preview1?

File: aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md (hunk 2 above)
Comment (scottaddie) [reply aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md]: The H2 is a great idea. My reservation with the H3 idea is that these attributes won't appear in the doc's "In this article" TOC on the right. Discoverability improves when used as H2s instead. Thoughts?

**Comment by NTaylorMullen** [comment]
> @NTaylorMullen I'd prefer to publish a doc for folks using Preview1. We took this approach with the new SPA templates and got some really great early feedback as a result. With that in mind, is this doc accurate for Preview1?

Alright, we'll need to make sure we get out a follow up once preview2 is released then 😄

File: aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+The `name` attribute is required. It indicates the name or the path of the partial view to be rendered. When a partial view name is provided, the [view discovery](xref:mvc/views/overview#view-discovery) process is initiated. That process is bypassed when an explicit path is provided.
+
+The following markup uses an explicit path, indicating that *_ProductPartial.cshtml* is to be loaded from the *Shared* folder. Using the [asp-for](#asp-for) attribute, a model is passed to the partial view for binding.
+
+[!code-cshtml[](samples/TagHelpersBuiltIn/Pages/Product.cshtml?name=snippet_Name)]
+
+## asp-for
+
+The `asp-for` attribute assigns an expression to be evaluated against the current model. The following markup loads *_ProductPartial.cshtml*, binding to it the model's `Product` property:
```
Comment (NTaylorMullen) [inline aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md]: Do we have any sections talking about the capabilities of `ModelExpressions` (it infers the `Model.` by default, you can override with `@`)? If so, we should link to that here. If not, we should have a section in the MVC TagHelper docs .

File: aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md (hunk 2 above)
Comment (Rick-Anderson) [reply aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md]: Keep them H2's but add the leading H2.

File: aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md (hunk 4 above)
Comment (scottaddie) [reply aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md]: See the following issue: https://github.com/aspnet/Docs/issues/231. Any samples you have would be greatly appreciated in that issue. Until that doc is completed, we have nothing that talks about ModelExpressions in detail.

**Comment by scottaddie** [comment]
@NTaylorMullen I've created an issue to update this doc for Preview2.

**Pull request merged**